### ISSUE TITLE
Make the package (widget) leaner

### ISSUE THREAD

**Issue by VladimirMikulic** [lifinance/widget#291]
For some time now at work our Next.js 14 front-end codebase has been almost unusable. Initial compilation took more than a minute on my i7 CPU, HMR not working requiring full page refresh, changes taking several seconds to even propagate for page to refresh etc, Next.js server running out of memory crashing etc.

For initial compilation we saw message by Next.js "compiling 30487" modules. After we removed LiFi widget (0 other changes), modules number dropped to ~5100. That's right, 25000 modules less, 600% smaller, initial compilation took only 10 seconds, got back HMR and changes take about 500ms to propagate.

I saw you have dependencies on UI libs so I presume they could be causing it and I think such a simple widget (from UI standpoint) could really be simplifed and shouldn't depend on so much stuff. I'm still amazed that a single widget has 6x more modules than our entire app and libs combined.

**Comment by chybisov** [comment]
@VladimirMikulic thank you for the feedback. Such many modules are only processed in the development mode and they are tree-shaken in the production build. We will take measures to improve DX and shrink them out in the development mode as well.

**Comment by VladimirMikulic** [comment]
@chybisov appreciate the response. Yes, I was referring to development mode for all of the issues mentioned above.

**Comment by RAMTO** [comment]
> [@VladimirMikulic](https://github.com/VladimirMikulic) thank you for the feedback. Such many modules are only processed in the development mode and they are tree-shaken in the production build. We will take measures to improve DX and shrink them out in the development mode as well.

Yes developing is quit slow, and a saw that wagmi is importing metamask/sdk which is quite huge as well. Can't wait to see more optimisations!

**Comment by RAMTO** [comment]
I analyzed my build and saw that Metamask SDK is still included (I do not use it in my project)

![Image](https://github.com/user-attachments/assets/837a53f3-25a7-4927-b01c-95e05199fa7a)

Is far as I see, metmask/sdk is a dependency of wagmi (this is not used in my project, it's installed cause of the lifi/widget)

![Image](https://github.com/user-attachments/assets/d6d76589-38b0-4e45-88d8-21819ce9c264)

On top of that when developing I am compiling 34735 modules... When not including the lifi/widget modules are considerably less.

Tried dynamic importing the widget, but still is heavy while developing. 

Any advice how to optimize my build for dev or prod? 

Thanks!

**Comment by chybisov** [comment]
@RAMTO as you can see on the screenshot metamask sdk is a separate chunk in your build and will be served to users only when needed and used. During development, it can indeed increase the build time, but it shouldn't affect it too much if you use modern tools like Vite or Next.js with Turbopack. What's your setup? 

We have these points in our roadmap and aim to optimize this further soon by making some functionality more modular and opt-in.

**Comment by RAMTO** [comment]
> [@RAMTO](https://github.com/RAMTO) as you can see on the screenshot metamask sdk is a separate chunk in your build and will be served to users only when needed and used. During development, it can indeed increase the build time, but it shouldn't affect it too much if you use modern tools like Vite or Next.js with Turbopack. What's your setup?
> 
> We have these points in our roadmap and aim to optimize this further soon by making some functionality more modular and opt-in.

I am using Next.js 14 with webpack, tried numerous config optimizations without any success, will appreciate some advice regarding that 🙏

**Comment by chybisov** [comment]
@RAMTO Have you tried enabling turbopack? https://nextjs.org/docs/app/api-reference/turbopack

**Comment by RAMTO** [comment]
I believe turbopack is a feature of Next.js 15, sadly my project is a legacy one on v14

**Comment by chybisov** [comment]
@RAMTO you can use it with v14 as well. See here https://nextjs.org/docs/14/architecture/turbopack

Also, please check if you use the best out of SWC compiler configuration https://nextjs.org/docs/14/architecture/nextjs-compiler

**Comment by RAMTO** [comment]
> [@RAMTO](https://github.com/RAMTO) you can use it with v14 as well. See here https://nextjs.org/docs/14/architecture/turbopack
> 
> Also, please check if you use the best out of SWC compiler configuration https://nextjs.org/docs/14/architecture/nextjs-compiler

Will check, thanks a lot! 🙏

**Comment by hoepeyemi** [comment]
> [@VladimirMikulic](https://github.com/VladimirMikulic) thank you for the feedback. Such many modules are only processed in the development mode and they are tree-shaken in the production build. We will take measures to improve DX and shrink them out in the development mode as well.

Is this open for contribution?
